services/src/main/java/org/keycloak/services/filters/CachingGZIPEncodingInterceptor.java

```diff
@@ -0,0 +1,90 @@
+package org.keycloak.services.filters;
+
+import org.apache.commons.io.IOUtils;
+import org.apache.commons.io.output.TeeOutputStream;
+import org.jboss.resteasy.plugins.interceptors.encoding.GZIPEncodingInterceptor;
+import org.jboss.resteasy.resteasy_jaxrs.i18n.LogMessages;
+import org.keycloak.common.Profile;
+
+import javax.annotation.Priority;
+import javax.ws.rs.Priorities;
+import javax.ws.rs.WebApplicationException;
+import javax.ws.rs.core.Context;
+import javax.ws.rs.core.HttpHeaders;
+import javax.ws.rs.core.UriInfo;
+import javax.ws.rs.ext.Provider;
+import javax.ws.rs.ext.WriterInterceptorContext;
+import java.io.ByteArrayOutputStream;
+import java.io.FileInputStream;
+import java.io.FileOutputStream;
+import java.io.IOException;
+import java.io.OutputStream;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.nio.file.Paths;
+
+@Provider
+@Priority(Priorities.ENTITY_CODER)
+public class CachingGZIPEncodingInterceptor extends GZIPEncodingInterceptor {
+
+    private final Path tmpDir;
+    @Context
+    private UriInfo uriInfo;
+
+    public CachingGZIPEncodingInterceptor() {
+        try {
+            this.tmpDir = Files.createTempDirectory("gzip-cache");
```

Comment:
Might be better to use tmp directory within the distribution rather than system tmp directory. We also need to purge the directory when the resource version changes.

Reply:
How do I know the version has changed? Seems that version is part of the request path, when the version changes the old versions are no longer needed, is that always true?

Isn't the `java.io.tmpdir` variable set to the tmp dir in the distribution? If not what is the way to create a temp folder, I couldn't find it in the code base.

Reply:
https://github.com/keycloak/keycloak/blob/master/common/src/main/java/org/keycloak/common/Version.java#L35

During startup this field is set to a randomly generated string specific for that version and that installation of KC. So if this doesn't match the version that was previously cached the old cached versions should be deleted. Perhaps the simplest would be to just add to a the folder `gzip-cache/Version.RESOURCES_VERSION` then make sure there is no other directories within gzip-cache than the current?